### PR TITLE
Upgrade to Rust 1.60

### PR DESCRIPTION
**Pull request by mhammond**
mozilla-central is moving to this, so I figured it can't hurt to see what the fallout is here - ~~apparently there is none!~~ but of course there is some.

### CODE REVIEW

**Comment by mhammond** [comment]
> it can't hurt to see what the fallout is here - apparently there is none!

doh - I forgot to run clippy :)

File: uniffi/src/ffi/rustbuffer.rs [hunk 1]
```diff
@@ -365,7 +365,11 @@ mod test {
         // We don't want to actually materialize a huge vec, so unsafety it is!
         // This won't cause problems because the contained items are Plain Old Data.
```
Comment (jplatte) [inline uniffi/src/ffi/rustbuffer.rs:366]: Have these tests ever been run through miri? I'm somewhat certain undef is not okay even for Plain Old Data, and you have to go through `MaybeUninit` instead.

Comment (bendk) [reply uniffi/src/ffi/rustbuffer.rs:366]: I don't think this should block the PR, but `MaybeUninit` does seem better to me.  I'll make a separate PR for that.

**Review by bendk — APPROVED** [review]
LGTM.  I like how each time we upgrade Rust, we get to clean up the code.

File: uniffi/src/ffi/rustbuffer.rs (hunk 1 above)
Comment (mhammond) [reply uniffi/src/ffi/rustbuffer.rs]: Ben ended up killing these tests in #1216

Comment (jplatte) [reply uniffi/src/ffi/rustbuffer.rs]: Yeah I noticed 🙂

**Pull request merged**